### ISSUE TITLE
[Composer\Downloader\TransportException] file could not be downloaded: php_network_getaddresses: getaddrinfo failed: Name or service not known

### ISSUE THREAD

**Issue by jaras2678** [composer/composer#6994]
Hi

No slash between domain and hash (https://gitlab.xx.pl61d8d4ced878a38328313aab96868e316c13c7c), error has occurred since version 1.6.2

My `composer.json`:

```json
{
  "name": "trimo/account-notification",
  "description": "",
  "minimum-stability": "dev",
  "license": "proprietary",
  "authors": [
    {
      "name": "xx xx",
      "email": "xx.xx@xx.pl"
    }
  ],
  "config": {
      "gitlab-token": {"gitlab.xxx.pl": "xx_xx"},
      "gitlab-domains": [
        "gitlab.xxx.pl"
      ]
  },
  "repositories": [
    {
      "type": "gitlab",
      "url": "git@gitlab.xxx.pl:trimo/lib.git"
    }
  ],
  "require": {
    "trimo/lib": "dev-master",
    "nesbot/carbon": "1.22.1",
    "swiftmailer/swiftmailer": "6.0.1"
  },
  "autoload": {
    "psr-4": {
      "Trimo\\": "src/"
    }
  }
}

```

When I run this command:

```
composer install
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)         
Package operations: 8 installs, 0 updates, 0 removals
  - Installing guzzlehttp/promises (dev-master 2e48ae6): Cloning 2e48ae638d from cache
  - Installing psr/http-message (dev-master f6561bf): Cloning f6561bf28d from cache
  - Installing guzzlehttp/psr7 (dev-master d2537c8): Cloning d2537c86fa from cache
  - Installing guzzlehttp/guzzle (6.3.0): Downloading (100%)         
  - Installing psr/log (dev-master 4ebe3a8): Cloning 4ebe3a8bf7 from cache
  - Installing monolog/monolog (1.23.0): Downloading (100%)         
  - Installing predis/predis (v1.1.1): Downloading (100%)         
  - Installing trimo/lib (dev-master 961d8d4): Cloning 961d8d4ced
    Failed to download trimo/lib from source: Failed to execute git clone --no-checkout 'git@gitlab.xx.pl:trimo/lib.git' '/home/src/vendor/trimo/lib' && cd '/home/src/vendor/trimo/lib' && git remote add composer 'git@gitlab.xx.pl:trimo/lib.git' && git fetch composer

Cloning into '/home/src/vendor/trimo/lib'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

    Now trying to download from dist
  - Installing trimo/lib (dev-master 961d8d4): Downloading (failed)       
Downloading (failed)       
Downloading (failed)       
                                                                                                                                                                                 
  [Composer\Downloader\TransportException]                                                                                                                                       
  The "https://gitlab.xx.pl61d8d4ced878a38328313aab96868e316c13c7c" file could not be downloaded: php_network_getaddresses: getaddrinfo failed: Name or service not known  
  failed to open stream: php_network_getaddresses: getaddrinfo failed: Name or service not known                                                                                 
                                                                                                                                                                                 

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...

```

No slash between domain and hash (https://gitlab.xx.pl61d8d4ced878a38328313aab96868e316c13c7c), error has occurred since version 1.6.2

**Comment by sjadema** [comment]
Can confirm that this is a problem since the 1.6.2 release.

**Comment by ymc-dabe** [comment]
I can confirm this as well: With 1.6.1 it works for me, but not with 1.6.2.
I looks like it was caused by commit e6114b2ca7b2eb75920fd03957070045a1ac1bc1. At least reverting this commit on top of tag 1.6.2 the error is gone.

**Comment by Tomassie91** [comment]
Issue is caused by the replacement parameter in the preg_replace for custom gitlab domains in `\Composer\Util\Url::updateDistReference()`. When the `$ref` parameter starts with a number, the ref gets concatenated with `$1`, causing an invalid group. e.g. 

```php
$ref = 65;
$replacement = '$1' . $ref;
var_dump($replacement); // $165, this is not a valid capture group in the REGEX pattern 
```

Solution is to encapsulate the group, see [PHP documentation](http://php.net/manual/en/function.preg-replace.php#refsect1-function.preg-replace-parameters).

**Comment by Seldaek** [comment]
Please use the latest snapshot for the fix until 1.6.3 comes out (`composer self-update --snapshot`)
